eth/fetcher/fetcher.go

```diff
@@ -648,7 +648,7 @@ func (f *Fetcher) insert(peer string, block *types.Block) {
 			return
 		}
 		// Quickly validate the header and propagate the block if it passes
-		switch err := f.validateBlock(block, parent); err {
+		switch err := f.verifyHeader(block.Header()); err {
```

Comment:
Looks like this changes the current mechanism which determines how 'deeply' we verify blocks/headers before propagation. Is this mentioned in the PR description? What are the consequences?